ui/app/src/components/GitManagement/PullDialog/PullDialogContainer.tsx

```diff
@@ -58,16 +78,15 @@ export function PullDialogContainer(props: PullFromRemoteDialogContainerProps) {
     isSubmitting = false
   } = props;
 
-  const [selectedBranch, setSelectedBranch] = useState('');
   const [selectedMergeStrategy, setSelectedMergeStrategy] = useState('');
   const { classes } = useStyles();
   const { uuid, id: siteId } = useActiveSite();
   const { username } = useActiveUser();
+  const [sandboxBranch, setSandboxBranch] = useState('');
+  const [sandboxBranchError, setSandboxBranchError] = useState<ApiResponse>(null);
```

Comment:
merge these into a _spread state_ and add a loading flag and implement it in the brach display area.